robotiq_joint_state_publisher/src/s_model_joint_states.cpp

```diff
@@ -0,0 +1,208 @@
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+/** @file s_model_joint_states.cpp
+ *  Subscribes to Robotiq state messages on "SModelRobotInput" topic, converts the data to joint values,
+ *  and publishes sensor_msgs/JointState messages on "joint_states" topic for Robotiq S-model.
+ * 
+ *  'rosrun robotiq_joint_state_publisher s_model_joint_states <gripper_prefix>'
+ * 
+ *  @author jack.thompson(at)utexas.edu
+ *  @maintainer karl.kruusamae(at)utexas.edu
+ */
+
+#include <vector>
+#include <string>
+#include <csignal>
+#include <ros/ros.h>
+#include <sensor_msgs/JointState.h>
+#include <robotiq_s_model_control/SModel_robot_input.h>
+#include <robotiq_s_model_control/SModel_robot_output.h>
+
+
+const int BASIC   = 0;
+const int PINCH   = 1;
+const int WIDE    = 2;
+const int SCISSOR = 3;
+
+const double DEG_TO_RAD = M_PI/180.0;
+
+/**
+ * Finger class for single Robotiq S-model finger.
+ */
+class Finger {
+ public:
+  Finger() { position = 0; }				///< Default constructor for creating Finger, position is set 0
+  Finger(int pos) { position = pos; }			///< Create Finger, position is @param pos
+  Finger(const Finger &f) { position = f.position; }	///< Create Finger, position is taken form @param Finger
+  inline double joint_1() const;			///< joint_1 value for Finger
+  inline double joint_2() const;			///< joint_2 value for Finger
+  inline double joint_3() const;			///< joint_3 value for Finger
+  int position;						///< Position of the Finger
+};
+
+/**
+ * Calculate joint_1 based on the Finger position.
+ */
+inline double Finger::joint_1() const {
+  if(0 <= position && position <= 140) return (70.0/148.0 * DEG_TO_RAD) * position;
+  else return 70.0 * DEG_TO_RAD;
+}
+
+/**
+ * Calculate joint_2 based on the Finger position.
+ */
+inline double Finger::joint_2() const {
+  if(0 <= position && position <= 140) return 0.0;
+  else if(140 < position && position <= 240) return (90.0/100.0 * DEG_TO_RAD) * (position-140.0);
+  else return 90.0 * DEG_TO_RAD;
+}
+
+/**
+ * Calculate joint_3 based on the Finger position.
+ */
+inline double Finger::joint_3() const {
+  if(0 <= position && position <= 140) return (-70.0/140.0 * DEG_TO_RAD) * position;
+  else return -55.0 * DEG_TO_RAD;
+}
+
+/**
+ * Robotiq S-model with three Fingers.
+ */
+class Robotiq3 {
+ public:
+  Robotiq3() { mode = BASIC; scissor = 137; joint_positions.resize(11, 0.0); }		///< Default constructor for creating Robotiq3
```

Comment:
The initialization of the vector can be done via [initializer list](http://en.cppreference.com/w/cpp/language/initializer_list) and the [fill constructor](http://www.cplusplus.com/reference/vector/vector/vector/).  Just an FYI, no need to change.